webapp/channels/src/components/permissions_gates/system_permission_gate/system_permission_gate.tsx

```diff
@@ -25,18 +25,15 @@ type Props = {
     children: React.ReactNode;
 };
 
-export default class SystemPermissionGate extends React.PureComponent<Props> {
-    public static defaultProps = {
-        invert: false,
-    };
-
-    render() {
-        if (this.props.hasPermission && !this.props.invert) {
-            return this.props.children;
-        }
-        if (!this.props.hasPermission && this.props.invert) {
-            return this.props.children;
-        }
-        return null;
+export default function SystemPermissionGate(props: Props) {
+    if (props.hasPermission && !props.invert) {
+        return props.children;
+    }
+    if (!props.hasPermission && props.invert) {
+        return props.children;
     }
+    return null;
 }
+SystemPermissionGate.defaultProps = {
```

Comment:
For function components, you wouldn't use defaultProps. The most common approach we use is to expand the props on the function and assign there the default. In this case...

```
const SystemPermissionGate = ({
   permissions,
   hasPermissions,
   invert = false,
   children, 
}: Props) => {
```